### DISCUSSION TITLE
Retrieve all successful rules

### DISCUSSION THREAD

**Discussion by good-times-coming** [microsoft/RulesEngine#350] [Q&A]
Is there a way to get the success events from all rules that succeed in a particular workflow?

```json
{
  "workflowName": "Validate",
  "rules": [
    {
      "ruleName": "Question_1",
      "successEvent": "1",
      "ruleExpressionType": "LambdaExpression",
      "expression": "input1.Answers[0] == 10"
    },
    {
      "ruleName": "Question_2",
      "successEvent": "2",
      "ruleExpressionType": "LambdaExpression",
      "expression": "input1.Answers[1] == 8"
    },
    {
      "ruleName": "Question_3",
      "successEvent": "3",
      "ruleExpressionType": "LambdaExpression",
      "expression": "input1.Answers[2] == 3"
    }
  ]
}
```
Given input:
`  {
    "Parameter": {
      "Answers": [10,8,3]
    },
    "InputRule": "input1"
  }`

I'd like to be able to get back an IEnumerable with values {1,2,3}, corresponding to the success events of each rule that passed.
Can I do anything to get a list of all the rules that pass, besides walking the whole RuleResultsTree?

**Comment by jmukherjee** [comment]
You can traverse RuleResultTree or use Linq. Each rule execution has a IsSuccess field either true/false.
May be this gives you some pointer
https://github.com/microsoft/RulesEngine/blob/3a5e31571f94ef8b6dabe7367cd0db558f35a979/src/RulesEngine/Extensions/ListofRuleResultTreeExtension.cs#L25
